jirate/jira_cli.py

```diff
@@ -476,19 +505,40 @@ def create_from_template(args):
         raise e
 
     for issue in template['issues']:
+        reserved_fields = ['subtasks']
+        required_fields = ['summary']
+        translate_fields = {'issue_type': 'issuetype'}
+
+        # These may be overridden
+        start_fields = {'issuetype': 'Task', 'project': args.project.project_name}
+
+        creation_fields = _parse_creation_args(issue, required_fields, reserved_fields, translate_fields, start_fields)
+
+        # Cache all metadata now (so we can debug subtask creation if needed.
+        issuetype = creation_fields['issuetype']
+        if issuetype not in metadata_by_type:
+            metadata_by_type[issuetype] = _metadata_by_type(args.project, issuetype)
+        if 'subtasks' in issue and issue['subtasks']:
+            metadata_by_type[_subtask] = _metadata_by_type(args.project, _subtask)
```

Comment:
Two related things here. They seem kind of a pain to fix for a case with... zero adoption, so I'd be fine leaving them as a known issue we can come back to in the future.

**issue:** In these two `_metadata_by_type()` calls we pass `args.project` even though we allow templates to set a different target project, which ends up in `creation_fields` and correctly used in the `create()` function. This will lead to fetching the wrong metadata.
**suggestion:** `issue_metadata()` doesn't allow passing a different project name so I guess we'd have to instantiate a new `JiraProject`? That or refactor `issue_metadata()`. 

**issue:** (second layer to the above) We allow having issues for multiple projects in the one template, but this cache doesn't differentiate between projects. e.g. if the template includes a Story for the FOO project and a Story for the BAR project, the FOO Story `metadata['fields']` will be used for both.
**suggestion:** Making this a dict of dicts would fix that -- something like `metadata_by_type[creation_fields['project']][issuetype]`